src/textual/widgets/_directory_tree.py

```diff
@@ -364,7 +364,9 @@ async def watch_path(self) -> None:
         """
         self.reset_node(self.root, str(self.path), DirEntry(self.PATH(self.path)))
         await self.reload()
-        self.move_cursor(self.root, animate=False)
+        if self.cursor_node is not None:
+            self.cursor_line = 0
```

Comment:
I'm surprised that you need to explicitly set this reactive here, as I'd expect that to be done by `move_cursor`? If they're not being kept in sync automatically, is that a bug?

Reply:
So if I, as a user, set `cursor_line`, it _doesn't_ move the cursor? I need to explicitly call `move_cursor`?

Reply:
It does move the cursor, but it doesn't change the scroll offset. Setting `cursor_line` will also update `cursor_node`.